### PR TITLE
Mojito library feature requests & updates 2023

### PR DESCRIPTION
**Pull request by allmywant**
1. Bump the Node packages & resolve all the security updates
2. Bump the GitHub unit test suite to run against later Node.JS versions: removed v12, added v18
3. Fix the error tracking to allow errors to show in the console for debugging
4. Fix a infinite loop bug in Utils.observeSelector function
5. Update the copyright notice to 2023
6. Address any ES Lint warnings or consider the rules in ES Lint
   Removed the following rules:

no-param-reassign
no-lonely-if
no-underscore-dangle
no-mixed-operators
no-useless-escape

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ function test()
 }
 
 function build()
-{
+{console.log('aa');
```
Comment (dapperdrop) [inline gulpfile.js]: @allmywant what is the reason for this `console.log`?

**Review by dapperdrop — CHANGES_REQUESTED** [review]

File: gulpfile.js (hunk 1 above)
Comment (allmywant) [reply gulpfile.js]: @dapperdrop 
That was the debug code I made and now I've removed it.

Comment (dapperdrop) [reply gulpfile.js]: Thanks @allmywant

**Review by dapperdrop — APPROVED** [review]

**Review by kingo55 — APPROVED** [review]
Nailed it, @allmywant - Thanks for this PR... will get this merged.

**Pull request merged**